planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java

```diff
@@ -132,9 +135,12 @@ private static Expression simplify(Expression initial) {
     seen.add(simplified);
     while (true) {
       simplified = simplifyOnce(simplified);
-      if (seen.contains(simplified) || seen.size() > 100) {
+      if (seen.contains(simplified)) {
         return simplified;
       }
+      if (seen.size() > 1000) {
+        throw new IllegalStateException("Infinite loop while simplifying expression " + initial);
+      }
```

Comment:
I was unable to compose an expression that triggers this as an actual infinite loop, though since it's only done at simplify time it's probably fine as a fail-safe.

Reply:
yeah, I'd like to leave it in just in case there's ever an unforeseen case where simplify logic goes off the rails